File: woodroid/src/main/java/me/gilo/woodroid/models/Order.java
package me.gilo.woodroid.models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;


public class Order {
    public int id;
    @SerializedName("order_number")
    public String orderNumber;
    @SerializedName("created_at")
    public String createdAt;
    @SerializedName("updated_at")
    public String updatedAt;
    @SerializedName("completed_at")
    public String completedAt;
    public String status;
    public String currency;
    public String total;
    public String subtotal;
    @SerializedName("total_line_items_quantity")
    public int totalLineItemsQuantity;
    @SerializedName("total_tax")
    public String totalTax;
    @SerializedName("total_shipping")
    public String totalShipping;
    @SerializedName("cart_tax")
    public String cartTax;
    @SerializedName("shipping_tax")
    public String shippingTax;
    @SerializedName("total_discount")
    public String totalDiscount;
    @SerializedName("shipping_methods")
    public String shippingMethods;
    @SerializedName("payment_details")
    public PaymentDetails paymentDetails;
    @SerializedName("billing")
    public BillingAddress billingAddress;
    @SerializedName("shipping")
    public ShippingAddress shippingAddress;
    public String note;
    @SerializedName("customer_ip")
    public String customerIp;
    @SerializedName("customer_user_agent")
    public String customerUserAgent;
    @SerializedName("customer_id")
    public Integer customerId;
    @SerializedName("view_order_url")
    public String viewOrderUrl;
    @SerializedName("line_items")
    public List<LineItem> lineItems = new ArrayList<LineItem>();
    @SerializedName("shipping_lines")
    public List<ShippingLine> shippingLines = new ArrayList<>();
    @SerializedName("tax_lines")
    public List<TaxLine> taxLines = new ArrayList<>();
    @SerializedName("fee_lines")
    public List<FeeLine> feeLines = new ArrayList<>();
    @SerializedName("coupon_lines")
    public List<Object> couponLines = new ArrayList<Object>();
    public Customer customer;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getOrderNumber() {
        return orderNumber;
    }

    public void setOrderNumber(String orderNumber) {
        this.orderNumber = orderNumber;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public String getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(String updatedAt) {
        this.updatedAt = updatedAt;
    }

    public String getCompletedAt() {
        return completedAt;
    }

    public void setCompletedAt(String completedAt) {
        this.completedAt = completedAt;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getCurrency() {
        return currency;
    }

    public void setCurrency(String currency) {
        this.currency = currency;
    }

    public String getTotal() {
        return total;
    }

    public void setTotal(String total) {
        this.total = total;
    }

    public String getSubtotal() {
        return subtotal;
    }

    public void setSubtotal(String subtotal) {
        this.subtotal = subtotal;
    }

    public Integer getTotalLineItemsQuantity() {
        return totalLineItemsQuantity;
    }

    public void setTotalLineItemsQuantity(Integer totalLineItemsQuantity) {
        this.totalLineItemsQuantity = totalLineItemsQuantity;
    }

    public String getTotalTax() {
        return totalTax;
    }

    public void setTotalTax(String totalTax) {
        this.totalTax = totalTax;
    }

    public String getTotalShipping() {
        return totalShipping;
    }

    public void setTotalShipping(String totalShipping) {
        this.totalShipping = totalShipping;
    }

    public String getCartTax() {
        return cartTax;
    }

    public void setCartTax(String cartTax) {
        this.cartTax = cartTax;
    }

    public String getShippingTax() {
        return shippingTax;
    }

    public void setShippingTax(String shippingTax) {
        this.shippingTax = shippingTax;
    }

    public String getTotalDiscount() {
        return totalDiscount;
    }

    public void setTotalDiscount(String totalDiscount) {
        this.totalDiscount = totalDiscount;
    }

    public String getShippingMethods() {
        return shippingMethods;
    }

    public void setShippingMethods(String shippingMethods) {
        this.shippingMethods = shippingMethods;
    }

    public PaymentDetails getPaymentDetails() {
        return paymentDetails;
    }

    public void setPaymentDetails(PaymentDetails paymentDetails) {
        this.paymentDetails = paymentDetails;
    }

    public BillingAddress getBillingAddress() {
        return billingAddress;
    }

    public void setBillingAddress(BillingAddress billingAddress) {
        this.billingAddress = billingAddress;
    }

    public ShippingAddress getShippingAddress() {
        return shippingAddress;
    }

    public void setShippingAddress(ShippingAddress shippingAddress) {
        this.shippingAddress = shippingAddress;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note = note;
    }

    public String getCustomerIp() {
        return customerIp;
    }

    public void setCustomerIp(String customerIp) {
        this.customerIp = customerIp;
    }

    public String getCustomerUserAgent() {
        return customerUserAgent;
    }

    public void setCustomerUserAgent(String customerUserAgent) {
        this.customerUserAgent = customerUserAgent;
    }

    public Integer getCustomerId() {
        return customerId;
    }

    public void setCustomerId(Integer customerId) {
        this.customerId = customerId;
    }

    public String getViewOrderUrl() {
        return viewOrderUrl;
    }

    public void setViewOrderUrl(String viewOrderUrl) {
        this.viewOrderUrl = viewOrderUrl;
    }

    public List<LineItem> getLineItems() {
        return lineItems;
    }

    public void setLineItems(List<LineItem> lineItems) {
        this.lineItems = lineItems;
    }

    public void addLineItem(LineItem lineItem) {
        lineItems.add(lineItem);

    }

    public List<ShippingLine> getShippingLines() {
        return shippingLines;
    }

    public void setShippingLines(List<ShippingLine> shippingLines) {
        this.shippingLines = shippingLines;
    }

    public List<TaxLine> getTaxLines() {
        return taxLines;
    }

    public void setTaxLines(List<TaxLine> taxLines) {
        this.taxLines = taxLines;
    }

    public List<FeeLine> getFeeLines() {
        return feeLines;
    }

    public void setFeeLines(List<FeeLine> feeLines) {
        this.feeLines = feeLines;
    }

    public List<Object> getCouponLines() {
        return couponLines;
    }

    public void setCouponLines(List<Object> couponLines) {
        this.couponLines = couponLines;
    }

    public Customer getCustomer() {
        return customer;
    }

    public void setCustomer(Customer customer) {
        this.customer = customer;
    }
}

File: woodroid/src/main/java/me/gilo/woodroid/models/Order.kt
package me.gilo.woodroid.models

import com.google.gson.annotations.SerializedName

import java.util.ArrayList
import java.util.Date


class Order {
    var id: Int = 0
    @SerializedName("number")
    var orderNumber: String
    @SerializedName("created_at")
    var createdAt: String

    @SerializedName("date_created")
    var dateCreated: Date

    @SerializedName("updated_at")
    var updatedAt: String
    @SerializedName("completed_at")
    var completedAt: String
    var status: String
    var currency: String
    var total: String
    var subtotal: String
    @SerializedName("total_line_items_quantity")
    var totalLineItemsQuantity: Int = 0
    @SerializedName("total_tax")
    var totalTax: String
    @SerializedName("total_shipping")
    var totalShipping: String
    @SerializedName("cart_tax")
    var cartTax: String
    @SerializedName("shipping_tax")
    var shippingTax: String
    @SerializedName("total_discount")
    var totalDiscount: String
    @SerializedName("shipping_methods")
    var shippingMethods: String
    @SerializedName("payment_details")
    var paymentDetails: PaymentDetails
    @SerializedName("billing")
    var billingAddress: BillingAddress
    @SerializedName("shipping")
    var shippingAddress: ShippingAddress
    var note: String
    @SerializedName("customer_ip")
    var customerIp: String
    @SerializedName("customer_user_agent")
    var customerUserAgent: String
    @SerializedName("customer_id")
    var customerId: Int? = null
    @SerializedName("view_order_url")
    var viewOrderUrl: String
    @SerializedName("line_items")
    var lineItems: MutableList<LineItem> = ArrayList()
    @SerializedName("shipping_lines")
    var shippingLines: List<ShippingLine> = ArrayList()
    @SerializedName("tax_lines")
    var taxLines: List<TaxLine> = ArrayList()
    @SerializedName("fee_lines")
    var feeLines: List<FeeLine> = ArrayList()
    @SerializedName("coupon_lines")
    var couponLines: List<Any> = ArrayList()
    var customer: Customer

    fun getId(): Int? {
        return id
    }

    fun setId(id: Int?) {
        this.id = id!!
    }

    fun getTotalLineItemsQuantity(): Int? {
        return totalLineItemsQuantity
    }

    fun setTotalLineItemsQuantity(totalLineItemsQuantity: Int?) {
        this.totalLineItemsQuantity = totalLineItemsQuantity!!
    }

    fun getLineItems(): List<LineItem> {
        return lineItems
    }

    fun setLineItems(lineItems: MutableList<LineItem>) {
        this.lineItems = lineItems
    }

    fun addLineItem(lineItem: LineItem) {
        lineItems.add(lineItem)

    }
}
